include/directx/d3d12shader.h

```diff
@@ -416,6 +608,10 @@ DECLARE_INTERFACE_(ID3D12ShaderReflection, IUnknown)
                                          _Out_opt_ UINT* pSizeZ) PURE;
 
     STDMETHOD_(UINT64, GetRequiresFlags)(THIS) PURE;
+
+    STDMETHOD_(BOOL, GetWaveSize)

```

Comment:
This kind of breaking change made sense with the FXC model, where you linked against and shipped a redist `D3DCompiler_xx.dll`, and also makes sense with a strictly-DXC model. But in a world where `D3DCompiler_47.dll` is inbox in Windows and implements the old interface, where DXC can implement a new one, I think I'd prefer to rev this in a non-breaking way.

We can do that by just adding `ID3D12ShaderReflection1`, which inherits from `ID3D12ShaderReflection` and adds the new method. The VTable of `ID3D12ShaderReflection1` is then identical to this new definition, except that apps can still query for the older one to avoid it being a source-breaking change for FXC consumers.